core/src/main/java/tc/oc/pgm/core/Core.java

```diff
@@ -160,6 +161,10 @@ public void markLeaked() {
     this.leaked = true;
   }
 
+  public Set<SelfIdentifyingFeatureDefinition> getModeList() {
```

Comment:
```suggestion
  public ImmutableSet<Mode> getModes() {
```

And make this change across the places with this set, it's a set of Modes, not a set of any type of feature definition, also just call it `getModes`, and the variable in the definition `modes`,  it's not a list to start with (it's a set) and that's also the naming used in the xml, so easiest to keep it consistent.

You also need to move this up the hierarchy in interfaces to `ModeChangeGoal`, replacing `isAffectedByModeChanges`